### PR TITLE
[testing] auto-replay captured streams

### PR DESCRIPTION
**Pull request by stas00**
This is a sync with the BigScience project https://github.com/bigscience-workshop/Megatron-DeepSpeed/pull/120

tldr; `CaptureStd*` interferes with `pytest` flags like `-sv` that should normally prints the std streams to the output. This PR fixes that.

Detailed: We noticed that when `CaptureStd` was used on the whole sub-process run it'd make it very difficult to debug problematic tests, since well it eats the output. So this PR changes the default behavior to replay the captured streams and adds a config arg to disable that.

This is a testing feature so backward compatibility shouldn't be an issue, and it will make things easier for us developers, rather than more difficult. And by default `pytest` captures/hides the std streams anyway.

@sgugger, @LysandreJik

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks for adding this!
